src/nvim/option.h

```diff
@@ -49,8 +56,8 @@ typedef struct {
                             ///< buffer-local option: global value
   idopt_T indir;            ///< global option: PV_NONE;
                             ///< local option: indirect option index
-  bool hidden;              ///< option is hidden, any attempt to set its value will be ignored.
-  bool immutable;           ///< option is immutable, trying to set its value will give an error.
+  bool hidden;              ///< option is hidden, cannot get/set its value
+  OptImmutable immutable;   ///< option immutability, see OptImmutable
```

Comment:
if this is going to be an enum, can "hidden" also be expressed in the same enum? Perhaps this should just be part of `OptTypeFlags` ?

```suggestion
  OptType2 hidde;   ///< option hidden and/or immutable.
```